web/api/v1/api.go

```diff
@@ -1793,16 +1792,14 @@ func (api *API) respond(w http.ResponseWriter, req *http.Request, data interface
 }
 
 func (api *API) negotiateCodec(req *http.Request, resp *Response) (Codec, error) {
-	for _, clause := range goautoneg.ParseAccept(req.Header.Get("Accept")) {
-		for _, codec := range api.codecs {
-			if codec.ContentType().Satisfies(clause) && codec.CanEncode(resp) {
-				return codec, nil
-			}
+	for _, codec := range api.codecs {
+		if codec.CanEncode(req, resp) {
+			return codec, nil
```

Comment:
This change breaks the content negotiation behaviour expected when the `Accept` header is present ([docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Content_negotiation#the_accept_header)).

`ParseAccept` returns the accepted content type clauses sorted from most-preferred to least-preferred, and so this loop previously selected the most highly preferred content type that was supported by a codec, and returned that codec.

With this change, we'll pick the first codec that supports any of the acceptable content types.

Reply:
Good catch, I changed it back so the outer loop iterates over the sorted content types.

Reply:
As I mentioned in https://github.com/prometheus/prometheus/pull/14622#discussion_r1726600154, I still believe that https://github.com/prometheus/prometheus/pull/14622/commits/f6f65a7446581899752a77400baa33294fa4d3b9 represents the best middle-ground workaround approach for addressing this.

Reply:
From the Slack discussion it seems like the community prefers to only use `Accept` header for codec negotiation. I am going to close the PR but thanks again for taking the time to give feedback!